make/utils.cmake

```diff
@@ -221,4 +221,46 @@ if (UNIX)
     append_property_string(TARGET ${target} LINK_FLAGS "${ldflags}")
   endfunction (set_preferred_base_start_and_end)
 
+  function (processor_and_compiler_support_avx out)
+    set(avx_prog "int main() {\
+                    asm volatile(\"vmovdqu32 \%ymm0,(\%rsp)\"); \
+                    return 0;
+                  }")
+    set(CMAKE_REQUIRED_FLAGS "-mavx")
+    check_c_source_compiles("${avx_prog}" compiler_supports_avx)
+    if (NOT compiler_supports_avx)
+      message(STATUS "WARNING: Compiler does not support AVX, skipping tests")
+    endif ()
+    if (compiler_supports_avx)
+      exec_program(cat ARGS "/proc/cpuinfo" OUTPUT_VARIABLE cpuinfo)
+      string(REGEX REPLACE "^.*(avx).*$" "\\1" grep_for_avx ${cpuinfo})
+      string(COMPARE EQUAL "avx" "${grep_for_avx}" proc_found_avx)
+      if (NOT proc_found_avx)
```

Comment:
I think you can replace all 3 lines above with `if (cpuinfo MATCHES "avx")` or `if (cpuinfo MATCHES "^flags.*avx")`.  Ditto below.

Reply:
Ok, I re-factored.